doc/src/content/xdocs/spec.xml

```diff
@@ -1155,8 +1165,8 @@
       <title>Parsing Canonical Form for Schemas</title>
 
       <p>One of the defining characteristics of Avro is that a reader
-      is assumed to have the "same" schema used by the writer of the
-      data the reader is reading.  This assumption leads to a data
+      must use the "same" schema used by the writer of the data in
```

Comment:
Maybe this is a good place for the "structurally identical" point above. We could further define "same" as "structurally identical", which I think is clear in context.